### PR TITLE
Apr 82 80 78 75

### PR DESCRIPTION
**Pull request by pabdi**
PR for JIRA tasks: APR-82, APR-80, APR-78, APR-75 APR-100 and APR-95
fixes some of validation on submit action and its error handling for form1, form2 and form3
sections default opening
adding progress bar during form1 load
increased img quality on PDF generation
add form1 acknowledgement page
fix some internal errors on tooltips

### CODE REVIEW

File: web_app/src/app/form1/page.tsx [hunk 1]
```diff
@@ -13,53 +12,80 @@ import { Step1Data, Step2Data, Step3Data, Step4Data } from '../interfaces';
...
     const [isExpanded, setIsExpanded] = useState<boolean>(false);
     const [isLoading, setIsLoading] = useState(false);
     const [message, setMessage] = useState('');
-    const [submitError, setSubmitError] = useState<boolean>(false);
 
+    const combineError = (errorMsg: string, newError: string) => {
+        if (errorMsg.startsWith("There are error in"))
+            return errorMsg + ", " + newError;
+        else
+            return "There are error in " + newError;
```
Comment (shaunmillargov) [inline web_app/src/app/form1/page.tsx:38]: Grammer is unusal here. Should this be "There is an error " or "There are errors "?

File: web_app/src/app/form1/steps/step1.tsx [hunk 2]
```diff
@@ -93,6 +100,30 @@ const Step1 = forwardRef((props: Props, ref) => {
         props.step1DatatoSend(step1Data);
     };
 
+    const [tooltipContent, setTooltipContent] = useState<{ [key: string]: JSX.Element | string }>({});
+
+    useEffect(() => {
+        setTooltipContent({
+            tooltipContent1: "Enter first 8 numbers with the dash. Don't enter the digit in the grey box. Prohibition numbers start with 00, 21, 30 or 40.",
```
Comment (shaunmillargov) [inline web_app/src/app/form1/steps/step1.tsx:107]: Enter "the" first 8 "numbers"
Don't enter the "digits" in the grey box.

File: web_app/src/app/form1/steps/step1.tsx [hunk 3]
```diff
@@ -93,6 +100,30 @@ const Step1 = forwardRef((props: Props, ref) => {
...
+                    height={145}
+                />
+            ),
+            tooltipContent3: (
+                <img src="/assets/images/Prohibition Period and Type.png"
+                    width={280}
+                    height={180}
+                    alt="Info" />),
+            tooltipContent4: (
+                <p>Enter first 8 numbers with the dash.Don&apos;t enter the digit in the grey box. Prohibition numbers start with 00, 21, 30 or 40.</p>
```
Comment (shaunmillargov) [inline web_app/src/app/form1/steps/step1.tsx:121]: Enter "the" first 8 numbers. 
"digits" in the grey box.

File: web_app/src/app/form2/page.tsx [hunk 4]
```diff
@@ -24,11 +23,17 @@ export default function Page() {
     const [cleanControlProhibitionNumber, setCleanControlProhibitionNumber] = useState('');
     const [validDriverLastName, setValidDriverLastName] = useState(false);
     const [message, setMessage] = useState('');
-    const [submitError, setSubmitError] = useState<boolean>(false);
 
+    const [tooltipContent, setTooltipContent] = useState<{ [key: string]: JSX.Element | string }>({});
+
+    useEffect(() => {
+        setTooltipContent({
+            tooltipContent1: "Enter first 8 numbers with the dash.Don't enter the digit in the grey box. Prohibition numbers start with 00, 21, 30 or 40.",
```
Comment (shaunmillargov) [inline web_app/src/app/form2/page.tsx:31]: Same as above

**Review by shaunmillargov — CHANGES_REQUESTED** [review]
See my comments regarding some suggested grammatical fixes. 
Otherwise looks good - Thank you.

File: web_app/src/app/form1/steps/step1.tsx (hunk 3 above)
Comment (pabdi) [reply web_app/src/app/form1/steps/step1.tsx:121]: these were copied from the text already there not by me but sure can make the changes

File: web_app/src/app/form2/page.tsx (hunk 4 above)
Comment (pabdi) [reply web_app/src/app/form2/page.tsx:31]: same as above response

File: web_app/src/app/form1/steps/step1.tsx (hunk 3 above)
Comment (pabdi) [reply web_app/src/app/form1/steps/step1.tsx:121]: I looked at this, the same phrase is used on the page and the tooltip in form1 and form3 changing it here in these 2 tooltip cases seems not correct if not changing others. So for this best approach is to reach out to the client if this is what they want or not. If the client agrees I'll make the change everywhere.

File: web_app/src/app/form1/page.tsx (hunk 1 above)
Comment (pabdi) [reply web_app/src/app/form1/page.tsx:38]: yes, agree will make the change

**Review by shaunmillargov — APPROVED** [review]

**Pull request merged**